mesonbuild/utils/universal.py

```diff
@@ -1050,6 +1050,8 @@ def default_prefix() -> str:
         return 'c:/'
     if is_haiku():
         return '/boot/system/non-packaged'
+    if 'HOMEBREW_PREFIX' in os.environ:
+        return os.environ['HOMEBREW_PREFIX']
```

Comment:
As someone who hasn't really used homebrew, just want to better understand the impact here. This variable is set when? If homebrew is installed, by auto-injecting into shell initialization files? Only when explicitly entering a homebrew shell environment?

Reply:
Homebrew install instructs the user to add eval "$(/opt/homebrew/bin/brew shellenv)" to their .zprofile, which sets this.

For belt-and braces, we could check /usr/local as well as $HOMEBREW_PREFIX, in case of weird/old installs?

Reply:
No, I think supporting it when explicitly opted in to probably makes quite a bit of sense.

Reply:
I am sorry, but this is wrong. I don't think anyone who installs Homebrew would expect that suddenly mesons default install prefix is changed to Homebrew, this makes no sense...

And if I remember correctly even Homebrew itself would complain about non-homebrew managed installs in its install directory...